lib/sedna/core/incremental_learning/incremental_learning.py

```diff
@@ -0,0 +1,136 @@
+            hem = self.config.get("hem_name") or "IBT"
+
+        if hem_parameters is None:
+            hem_parameters = {}
+        self.hard_example_mining_algorithm = ClassFactory.get_cls(
+            ClassType.HEM, hem)(**hem_parameters)
+
+    def train(self, train_data,
+              valid_data=None,
+              post_process=None,
+              **kwargs):
+        callback_func = None
+        if post_process is not None:
+            callback_func = ClassFactory.get_cls(
+                ClassType.CALLBACK, post_process)
+
+        res = self.estimator.train(
+            train_data=train_data, valid_data=valid_data, **kwargs)
+        model_paths = self.estimator.save(self.model_path)
+        task_info_res = self.estimator.model_info(
+            model_paths, result=res, relpath=self.config.data_path_prefix)
+        self.report_task_info(
+            None, K8sResourceKindStatus.COMPLETED.value, task_info_res)
+        return callback_func(
+            self.estimator) if callback_func else self.estimator
+
+    def inference(self, data=None, post_process=None, **kwargs):
+        if not self.estimator.has_load:
+            self.estimator.load(self.model_path)
+
+        callback_func = None
+        if callable(post_process):
+            callback_func = post_process
+        elif post_process is not None:
+            callback_func = ClassFactory.get_cls(
+                ClassType.CALLBACK, post_process)
+        res = self.estimator.predict(data, **kwargs)
+        rsl = callback_func(deepcopy(res)) if callback_func else res
+        is_hard_example = False
+
+        if self.hard_example_mining_algorithm:
+            is_hard_example = self.hard_example_mining_algorithm(rsl)
+        return res, rsl, is_hard_example
+
+    def evaluate(self, data, post_process=None, **kwargs):
+        callback_func = None
+        if callable(post_process):
+            callback_func = post_process
+        elif post_process:
+            callback_func = ClassFactory.get_cls(
+                ClassType.CALLBACK, post_process)
+        final_res = []
+        if self.model_urls:
+            for model_url in self.model_urls.split(";"):
+                if not model_url.strip():
+                    continue
+                self.estimator.model_save_path = model_url
+                res = self.estimator.evaluate(
+                    data=data, model_path=model_url, **kwargs)
+                if callback_func:
+                    res = callback_func(res)
+                self.log.info(f"Evaluation with {model_url} : {res} ")
+                task_info_res = self.estimator.model_info(
+                    model_url, result=res,
+                    relpath=self.config.data_path_prefix)
+                if isinstance(task_info_res, (list, tuple)
+                              ) and len(task_info_res):
+                    task_info_res = list(task_info_res)[0]
+                final_res.append(task_info_res)
+        else:
```

Comment:
this else branch can be merge into if branch

Reply:
fix